### DISCUSSION TITLE
Syncplay x play-with-mpv

### DISCUSSION THREAD

**Discussion by Lyapsus** [Syncplay/syncplay#489] [Archive]
Is that any real to add to Syncplay playlist any video opened with https://github.com/Thann/play-with-mpv automatically?
Maybe I miss some mpv kung-fu but it would be pretty useful to spread Syncplay's functionality beyond local video files.

**Comment by Et0h** [comment]
Support for this has not been explicitly added to Syncplay.

**Comment by Lyapsus** [comment]
Not sure if I understood you right. Actually I can put a link to e.g. Youtube video in Syncplay playlist manually, but why not to automate it little further?

**Comment by Et0h** [comment]
People can already add items (including Youtube videos) to the Syncplay playlist from within mpv using the /qa command, and in Syncplay 1.7.0 you will be able to add and switch to the item with the /qas command. It might be possible to modify play-with-mpv to make use of that, but I'm not sure as I am not familiar with play-with-mpv and do not use it myself.

You ask the question "why not" add the functionality you request to Syncplay. The answer to that question is usually "because it would add to the complexity of Syncplay and require developer resources to implement and maintain".

In most cases features are added by volunteers who would regularly use the feature, and therefore have a strong motivation both to implement it and to test it as part of that usage and make changes to Syncplay (and any relevant third-party software) necessary to ensure it continues to work.

Given that third-party software can change, adding anything which involves the interaction between two pieces of third-party code (play-with-mpv and mpv itself) significantly increases the chance that at some point those projects might change something which would break the implementation, requiring an unpredictable amount of resources to fix (and in some cases requiring upstream fixes in the software itself).

Furthermore, this feature could require testing and maintenance across every single operating system Syncplay supports (Windows, Linux, *BSD, macOS) and given that each of these has multiple versions and distros (and Syncplay developer often only use one OS) this could be quite the development burden.

As noted at https://github.com/Syncplay/syncplay/issues/315 "Unnecessary complexity can lead to unpredictable behaviour, features people never use, and a harder to manage (and therefore more likely to be buggy/unreliable) code base. We want to cover the main uses cases, not every possible circumstance (especially where there are reasonable workarounds)."

P.S. This is a feature request rather than a bug, so has been moved from issues to discussions.
